src/python/pants/backend/python/dependency_inference/rules.py

```diff
@@ -518,11 +568,26 @@ async def resolve_parsed_dependencies(
             for imported_module in parsed_imports
         )
         resolve_results = _get_imports_info(
-            address=request.field_set.address,
+            address=req_address,
             owners_per_import=owners_per_import,
             parsed_imports=parsed_imports,
+        )
+        resolve_results = await _disambiguate_or_mark_unowned(
+            address=req_address,
+            resolve_results=resolve_results,
             explicitly_provided_deps=explicitly_provided_deps,
         )
+        for import_name, resolve_result in resolve_results.items():
+            if resolve_result.status in [
+                ImportOwnerResult.unambiguous, ImportOwnerResult.disambiguated,
+            ]:
+                continue
```

Comment:
I think you can also exclude `unownable` and `unowned`, since those are definitive statements about the imports